### PR TITLE
Fix some nit for kubelet

### PR DESCRIPTION
**Pull request by yuzhiquan**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
- error string should not be capitalized
- fix broken links by https
- fix potential nil pointer
- comment should have function name as prefix
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```
/triage accepted
/priority backlog

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_manager.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ type KubeGenericRuntime interface {
 
 // LegacyLogProvider gives the ability to use unsupported docker log drivers (e.g. journald)
 type LegacyLogProvider interface {
-	// Get the last few lines of the logs for a specific container.
+	// GetContainerLogTail get the last few lines of the logs for a specific container.
```
Comment (shubheksha) [inline pkg/kubelet/kuberuntime/kuberuntime_manager.go:155]:
```suggestion
	// GetContainerLogTail gets the last few lines of the logs for a specific container.
```

File: pkg/kubelet/kuberuntime/kuberuntime_container.go [hunk 2]
```diff
@@ -658,7 +658,7 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 			"containerName", containerName, "containerID", containerID.String(), "gracePeriod", gracePeriod)
 	}
 
-	klog.V(2).InfoS("Killing container with a grace period override", "pod", klog.KObj(pod), "podUID", pod.UID,
+	klog.V(2).InfoS("Killing container with a grace period", "pod", klog.KObj(pod), "podUID", pod.UID,
```
Comment (shubheksha) [inline pkg/kubelet/kuberuntime/kuberuntime_container.go:661]: Why have we removed the word "override" here?

File: pkg/kubelet/kubelet_pods.go [hunk 3]
```diff
@@ -1156,7 +1156,7 @@ type PodKiller interface {
 	// PerformPodKillingWork performs the actual pod killing work via calling CRI
 	// It returns after its Close() func is called and all outstanding pod killing requests are served
 	PerformPodKillingWork()
-	// After Close() is called, this pod killer wouldn't accept any more pod killing requests
+	// Close is called, then this pod killer wouldn't accept any more pod killing requests
```
Comment (shubheksha) [inline pkg/kubelet/kubelet_pods.go:1159]:
```suggestion
	// Close ensures that after it's called, this pod killer wouldn't accept any more pod killing requests
```

File: pkg/kubelet/container/runtime.go [hunk 4]
```diff
@@ -112,7 +112,7 @@ type Runtime interface {
 	// stream the log. Set 'follow' to false and specify the number of lines (e.g.
 	// "100" or "all") to tail the log.
 	GetContainerLogs(ctx context.Context, pod *v1.Pod, containerID ContainerID, logOptions *v1.PodLogOptions, stdout, stderr io.Writer) (err error)
-	// Delete a container. If the container is still running, an error is returned.
+	// DeleteContainer delete a container. If the container is still running, an error is returned.
```
Comment (shubheksha) [inline pkg/kubelet/container/runtime.go:115]:
```suggestion
	// DeleteContainer deletes a container. If the container is still running, an error is returned.
```

File: cmd/kubelet/app/server.go [hunk 5]
```diff
@@ -500,11 +500,11 @@ func getReservedCPUs(machineInfo *cadvisorapi.MachineInfo, cpus string) (cpuset.
 
 	topo, err := topology.Discover(machineInfo)
 	if err != nil {
-		return emptyCPUSet, fmt.Errorf("Unable to discover CPU topology info: %s", err)
+		return emptyCPUSet, fmt.Errorf("unable to discover CPU topology info: %s", err)
 	}
 	reservedCPUSet, err := cpuset.Parse(cpus)
 	if err != nil {
-		return emptyCPUSet, fmt.Errorf("Unable to parse reserved-cpus list: %s", err)
+		return emptyCPUSet, fmt.Errorf("unable to parse reserved-cpus list: %s", err)
```
Comment (shubheksha) [inline cmd/kubelet/app/server.go:507]: We should capitalise the first letter of the error message as per the migration guide: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#remove-string-formatting-from-log-message

I'm curious about where we've mentioned that error messages shouldn't be captalised as we need to fix it :)

File: cmd/kubelet/app/options/options.go [hunk 6]
```diff
@@ -528,8 +528,8 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
...
+	fs.Var(cliflag.NewMapStringString(&c.SystemReserved), "system-reserved", "A set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=500Mi,ephemeral-storage=1Gi) pairs that describe resources reserved for non-kubernetes components. Currently only cpu and memory are supported. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ for more detail. [default=none]")
+	fs.Var(cliflag.NewMapStringString(&c.KubeReserved), "kube-reserved", "A set of ResourceName=ResourceQuantity (e.g. cpu=200m,memory=500Mi,ephemeral-storage=1Gi) pairs that describe resources reserved for kubernetes system components. Currently cpu, memory and local ephemeral storage for root file system are supported. See https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ for more detail. [default=none]")
```
Comment (shubheksha) [inline cmd/kubelet/app/options/options.go:532]: Not sure why this URL has been changed as the current one leads to the right documentation page. Could you please elaborate?

**Review by shubheksha — CHANGES_REQUESTED** [review]

File: cmd/kubelet/app/server.go (hunk 5 above)
Comment (yuzhiquan) [reply cmd/kubelet/app/server.go:507]: That document is  for klog style, but here may be golang style, and as golang code style suugested here: https://github.com/golang/go/wiki/CodeReviewComments#error-strings, error strings should not be capitalized.

File: cmd/kubelet/app/options/options.go (hunk 6 above)
Comment (yuzhiquan) [reply cmd/kubelet/app/options/options.go:532]: Old url: http://kubernetes.io/docs/user-guide/compute-resources is unsafety HTTP, and auto redirect to new HTTPS url.

File: pkg/kubelet/kuberuntime/kuberuntime_container.go (hunk 2 above)
Comment (yuzhiquan) [reply pkg/kubelet/kuberuntime/kuberuntime_container.go:661]: gracePeriodOverride logic is in  https://github.com/kubernetes/kubernetes/blob/9809038728c56e646e05430904fb7e01e89ce473/pkg/kubelet/kuberuntime/kuberuntime_container.go#L657, i think this line is not override.

**Comment by yuzhiquan** [comment]
ping @mrunalp

**Review by mrunalp — APPROVED** [review]

**Comment by pacoxu** [comment]
/test pull-kubernetes-node-e2e-containerd 
for CI hang

**Comment by pacoxu** [comment]
/test pull-kubernetes-node-e2e-containerd 
for CI hang

**Pull request merged**